extension.js

```diff
@@ -273,27 +272,45 @@ async function build(config, componentPath, server) {
 		}
 
 		// Build
-		const [command, ...args] = shellQuote.parse(config.buildCommand);
+		try {
+			logger.info(`Building Next.js application at ${componentPath}`);
+
+			const timerStart = performance.now();
+
+			const stdout = [];
+			const stderr = [];
 
-		const timerStart = performance.now();
+			const buildProcess = spawn(config.buildCommand, [], {
+				shell: true,
+				cwd: componentPath,
+				stdio: ['ignore', 'pipe', 'pipe'],
+			});
 
-		const { stdout, stderr, status, error } = spawnSync(command, args, {
-			cwd: componentPath,
-			encoding: 'utf-8',
-		});
+			buildProcess.stdout.on('data', c => stdout.push(c));
```

Comment:
Would it be better to immediately output these to the logger (on each data event), for people that are tailing/monitoring their logs and just don't have the patience for the build to finish to see the progress?

Reply:
Since we've switched this to execute async, my worry is that it would get clobbered with other logs and become unreadable